### PR TITLE
Don't use anonymous namespace in header.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21790 Don't use anonymous namespace in header.**

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D15827311](https://our.internmc.facebook.com/intern/diff/D15827311)

### CODE REVIEW

**Comment by ezyang** [comment]
cc @kmatzen

File: aten/src/ATen/core/PhiloxRNGEngine.h [hunk 1]
```diff
@@ -21,7 +21,7 @@
 namespace at {
 
 // typedefs for holding vector data
-namespace {
+namespace detail {
```
Comment (colesbury) [inline aten/src/ATen/core/PhiloxRNGEngine.h]: There's still a comment "// anonymous namespace" below

File: aten/src/ATen/core/PhiloxRNGEngine.h [hunk 2]
```diff
@@ -21,7 +21,7 @@
...
 
 typedef at::detail::Array<uint32_t, 4> UINT4;
```
Comment (colesbury) [inline aten/src/ATen/core/PhiloxRNGEngine.h]: It looks like these are only used in `philox_engine` so maybe they should just be placed in `class philox_engine`, but this is fine too

**Review by syed-ahmed — APPROVED** [review]

File: aten/src/ATen/core/PhiloxRNGEngine.h (hunk 2 above)
Comment (ezyang) [reply aten/src/ATen/core/PhiloxRNGEngine.h]: You're right, but I am a lazy man :)

**Pull request closed without merging**